libbeat/idxmgmt/idxmgmt.go

```diff
@@ -57,25 +51,43 @@ type Supporter interface {
 
 	// Manager creates a new manager that can be used to execute the required steps
 	// for initializing an index, ILM policies, and write aliases.
-	Manager(client ESClient, assets Asseter) Manager
+	Manager(client Client, assets Asseter) Manager
 }
 
 // Asseter provides access to beats assets required to load the template.
 type Asseter interface {
 	Fields(name string) []byte
 }
 
-// ESClient defines the minimal interface required for the index manager to
+// Client defines the minimal interface required for the index manager to
 // prepare an index.
-type ESClient interface {
+type Client interface {
 	Request(method, path string, pipeline string, params map[string]string, body interface{}) (int, []byte, error)
 	GetVersion() common.Version
 }
```

Comment:
I don't care much about the name here.

fyi; I named this ESClient, cause the `Request` method mimics the `libbeat/output/elasticsearch.Client` type, which is legacy and should be removed in the future.

Maybe it would have been better to pass a more rich but abstract type to Manager that doesn't indicated that we run an HTTP request, yet we need it for the template :/

Reply:
I was thinking about rewriting and passing in a more abstract type, but then also didn't want to grow this PR too much and mix too many things. 

The renaming is an actual leftover, I can revert to `ESClient`.